### PR TITLE
Added tests for null key handling in DictionaryExtensions

### PR DESCRIPTION
**Pull request by gmottajr**
This pull request introduces the following changes to the DictionaryExtensions tests:

Test for Null Key Handling: Although an attempt was made to test how the AddMany method handles null keys when the key type is nullable (string?), it was not possible to implement this test. C# dictionaries do not allow null as a key, even for nullable key types like string?, because of the notnull constraint on the TKey parameter in the Dictionary class. The notnull constraint prevents nullable types like string? from being used as dictionary keys, and as a result, this test could not be completed due to the C# language limitation.
Test for Nullable Value Types: Included tests to verify that nullable value types (such as int?) are correctly handled by the AddMany method, ensuring that null can be assigned as a value.
Test for Duplicate Keys: Added a test case to check that an ArgumentException is thrown when attempting to add duplicate keys to the dictionary using AddMany. 
Edge Case Coverage: Ensured edge cases, such as adding large collections of key-value pairs, are properly handled.

**Motivation and Context:**
    The existing tests did not adequately cover scenarios involving null keys or nullable value types. This change is necessary to ensure that the AddMany method works as expected in edge cases, improving the overall test coverage.

The additional checks for duplicate keys ensure that the method behaves correctly when invalid inputs (such as duplicate keys) are provided, preventing potential runtime errors.

- [X] I have performed a self-review of my code
- [X] My code follows the style guidelines of this project
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Comments in areas I changed are up to date
- [X] I have added comments to hard-to-understand areas of my code
- [X] I have made corresponding changes to the README.md

### CODE REVIEW

File: Utilities.Tests/Extensions/DictionaryExtensionsTests.cs [hunk 1]
```diff
@@ -33,4 +33,103 @@ public void AddMany_ShouldAddAllKeyValuePairs()
...
+            enumerable.Add((i, "Value" + i));
+        }
+
+        dictionary.AddMany(enumerable);
+
+        dictionary.Should().HaveCount(10000);
+        dictionary[9999].Should().Be("Value9999");
+    }
+
+
```
Comment (siriak) [inline Utilities.Tests/Extensions/DictionaryExtensionsTests.cs:134]:
```suggestion
```
